activiti-cloud-services-org/activiti-cloud-services-org-service/src/main/java/org/activiti/cloud/services/organization/validation/ProcessModelValidator.java

```diff
@@ -94,11 +112,63 @@ public void validateModelContent(byte[] bytes) {
         }
     }
 
-    private List<ValidationError> validateTasksAssignedUser(BpmnModel bpmnModel) {
-        return bpmnModel.getProcesses().stream()
+    private List<ValidationError> validateServiceTasksImplementation(BpmnModel bpmnModel,
+                                                                     ValidationContext validationContext) {
+        List<String> availableImplementations = getAvailableImplementations(validationContext);
+
+        return getTasksStream(bpmnModel,
+                              ServiceTask.class)
+                .map(serviceTask -> validateServiceTaskImplementation(serviceTask,
+                                                                      availableImplementations))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+    }
+
+    private <T extends Task> Stream<T> getTasksStream(BpmnModel bpmnModel,
+                                                      Class<T> taskType) {
+        return bpmnModel.getProcesses()
+                .stream()
                 .flatMap(process -> process.getFlowElements().stream())
-                .filter(flowElements -> flowElements.getClass().isAssignableFrom(UserTask.class))
-                .map(UserTask.class::cast)
+                .filter(element -> taskType.isAssignableFrom(element.getClass()))
+                .map(taskType::cast);
+    }
+
+    private List<String> getAvailableImplementations(ValidationContext validationContext) {
+        return validationContext.getAvailableModels()
+                .stream()
+                .filter(model -> ConnectorModelType.NAME.equals(model.getType()))
+                .map(this::concatImplementationAndAction)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+    }
+
+    private List<String> concatImplementationAndAction(Model model) {
+        return connectorModelContentConverter.convertToModelContent(model.getContent().getBytes())
+                .filter(Objects::nonNull)
```

Comment:
This is not best way to handle the Optional returned by a method :)
Taking in consideration that also model.getContent() can be null, I think you can re-write it something like:
```
return Optional.ofNullable(model.getContent())
                .map(String::getBytes)
                .flatMap(connectorModelContentConverter::convertToModelContent)
                .map(ConnectorModelContent::getActions)
                .map(Map::values)
                .map(Collection::stream)
                .orElseGet(Stream::empty)
                .map(connectorModelAction -> StringUtils.isEmpty(connectorModelAction.getName()) ? model.getName() : model.getName() + "." + connectorModelAction.getName())
                .collect(Collectors.toList());
```